Yoast/wordpress-seo
# Adapt matchKeywordInSubheadings for Japanese

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Adapts `matchKeywordInSubheadings` research for Japanese.

## Relevant technical choices:

* The mechanism that calculates `percentWordMatches` in `findWordFormsInString` still makes sense for Japanese. Hence no changes/adaptation is made for the formula.
````
Math.round( foundNumberOfWordsInKeyphrase / totalWordNumberInKeyphrase * 100 )
````



## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Run `yarn test` and make sure that everything passes.

Test in Docker/WordPress env:
* Enable `JAPANESE_SUPPORT` by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', true );` the in `basic-wordpress-config.php` file
* Set your site to Japanese (日本語)
* Create a post (you can use this generator https://generator.lorem-ipsum.info/_japanese)
* Add this word as the keyphrase: 低照 (for now we can only test keyphrase that doesn't need segmentation)
* Add subheading(s) and test Keyphrase in Subheadings assessment using[ this criteria](https://github.com/Yoast/javascript/wiki/Scoring-SEO-analysis#5-keyphrase-in-subheadings)

For example:
green bullet, keyword is used in 1 out of 3 subheadings (= 33%):
小見出し内キーフレーズ: H2 または H3 小見出しのうち1個が文章のトピックを反映しています。いいですね !

green bullet, keyword is used in 2 out of 3 subheadings:
小見出し内キーフレーズ: H2 または H3 小見出しのうち2個が文章のトピックを反映しています。いいですね ! 

red bullet, keyword is used in 3 out of 3 subheadings:
小見出しのキーフレーズ: H2 または H3 小見出しの75%以上が文章のトピックを反映していますが、これは多すぎます。最適化しすぎないでください。

red bullet, keyword doesn’t match any of the subheadings:
小見出しのキーフレーズ: H2 または H3 レベルの小見出しで、もっとキーフレーズまたは類義語を使用してください。

* Save your changes

* Disable `JAPANESE_SUPPORT` by adding `define( 'YOAST_SEO_JAPANESE_SUPPORT', false );` the in `basic-wordpress-config.php` file
* Confirm that the Keyphrase in Subheadings assessment give red bullet and a feedback something along this line: `Use more key phrases or synonyms in H2 or H3 level subheadings.` Or `小見出しのキーフレーズ: H2 または H3 レベルの小見出しで、もっとキーフレーズまたは類義語を使用してください。`

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

* Test this step `Test in Docker/WordPress env`

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Other environments


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1102
